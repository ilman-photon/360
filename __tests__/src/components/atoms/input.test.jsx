import { render } from "@testing-library/react";

import StyledInput from "../../../../src/components/atoms/Input/input";
import "@testing-library/jest-dom";

describe("StyledInput", () => {
  it("renders input components with type text", () => {
    const { container } = render(<StyledInput type="text" />);
    const input = container.getElementsByClassName("custom-input");
    expect(input).toBeTruthy();
  });

  it("renders input components with type password", () => {
    const { container } = render(
      <StyledInput
        type="password"
        variant="filled"
        label="Text"
        adorment={true}
      />
    );
    const input = container.getElementsByClassName("custom-input");
    expect(input).toBeTruthy();
  });

  it("renders input components with type dob", () => {
    const { container } = render(
      <StyledInput
        type="dob"
        variant="outlined"
        label="Dates"
        adorment={false}
      />
    );
    const input = container.getElementsByClassName("custom-input");
    expect(input).toBeTruthy();
  });
});
