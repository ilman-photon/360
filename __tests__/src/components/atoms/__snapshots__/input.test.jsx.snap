// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInput renders input components with type text 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root custom-input css-1tsra5p-MuiFormControl-root-MuiTextField-root"
    tabindex="0"
    withicon="true"
  >
    <div
      aria-hidden="true"
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ye1xzc-MuiInputBase-root-MuiOutlinedInput-root"
      tabindex="-1"
    >
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        placeholder=""
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-hdw1oc"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
